File: SSServer/src/main/java/com/prose/crhen/SSServer/db/VolumeRepository.java
package com.prose.crhen.SSServer.db;

import com.prose.crhen.SSServer.model.Server;
import com.prose.crhen.SSServer.model.Volume;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface VolumeRepository extends JpaRepository<Volume, Long> {

    Volume findByName(String name);

    List<Volume> findByServer(Server server);
}

File: SSKotlinServer/src/main/kotlin/com/prose/crhen/SSKotlinServer/SSKotlinServer/db/VolumeRepository.kt
package com.prose.crhen.SSKotlinServer.SSKotlinServer.db

import com.prose.crhen.SSKotlinServer.SSKotlinServer.model.Server
import com.prose.crhen.SSKotlinServer.SSKotlinServer.model.Volume
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface VolumeRepository: JpaRepository<Volume, Long> {
    fun findByName(name: String): Volume

    fun findByServer(server: Server): List<Volume>

    fun findVolumeByNameAndServer(name: String, server: Server): Volume
}
